assets/javascripts/discourse/connectors/post-text-buttons/ai-helper-options-menu.gjs

```diff
@@ -50,8 +52,62 @@ export default class AIHelperOptionsMenu extends Component {
 
   @tracked _activeAIRequest = null;
 
+  highlightSelectedText() {
+    const postId = this.args.outletArgs.data.quoteState.postId;
+    const postElement = document.querySelector(
+      `article[data-post-id='${postId}']`
+    );
+
+    if (!postElement) {
+      return;
+    }
+
+    this.originalPostHTML = postElement.innerHTML;
+    this.selectedText = this.args.outletArgs.data.quoteState.buffer;
+
+    const selection = window.getSelection();
+    if (!selection.rangeCount) {
+      return;
+    }
+
+    const range = selection.getRangeAt(0);
+    const originalContents = range.extractContents();
+
+    const highlight = document.createElement("span");
+    highlight.classList.add("ai-helper-highlighted-selection");
+    highlight.appendChild(originalContents.cloneNode(true));
```

Comment:
So for multiple nodes, this adds one span element around all the nodes? It might be helpful to wrap each node in a highlighted span, it might avoid line break issues... not sure.